### ISSUE TITLE
Implement resource_update

### ISSUE THREAD

**Issue by florianm** [ropensci/ckanr#11]
I have an R object, e.g. a data.frame (to become a CSV table), a ggplot2 object (to become a PDF figure), a text string (to become a TXT file), and a write-permitted CKAN API key for a CKAN instance.

I want to upload each object to replace an existing CKAN resource, with or without first writing the objects to files.

This should work both locally and when used e.g. inside an RShiny running on an RShiny server.

This is the desired functionality in Python (my own code):
```{python}
# /usr/bin/python
import os, sys, requests, json
from datetime import datetime
env.use_ssh_config = True

DC = "http://my-ckan-instance.org/"

def resource_show(resource_id, api_url="{0}api/3/action/".format(DC)):
    """Return a JSON dictionary of dataset details for a given dataset id.

    :param dataset_id: A CKAN dataset id
    :param api_url: A live CKAN API URL, default: "<DC>/api/3/action/"

    :return: a JSON dictionary of dataset details
    """
    r = requests.get("{0}resource_show?id={1}".format(api_url, resource_id))
    if r.status_code == 200:
      return json.loads(r.content)["result"]
    else:
      return None 
      
def resource_update(filedir, res_id, filepath, 
  api_url="{0}api/3/action/".format(DC), api_key=None):
  """Update the file attachment of a given resource ID.

  :param res_id: The resource ID of an existing resource
  :param filepath: The path to a local file to be uploaded
  :param api_url: A live CKAN API URL, default: "<DC>/api/3/action/"
  :param api_key: A write-privileged API key (sensitive data)

  :return: None
  """
  res = resource_show(res_id)
  res["state"] = "active"
  res["last_modified"] = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
  
  if os.path.isfile(os.path.join(filedir, filepath)):
    r = requests.post("{0}resource_update".format(api_url),
              #data={"id": res_id},
              data=res,
              headers={"Authorization": api_key},
              files=[('upload', file(os.path.join(filedir, filepath)))])
    print("Uploaded {0}".format(filepath))
  else:
    print("File {0} not found, skipping upload".format(filepath))
    

```

**Comment by sckott** [comment]
hi @florianm - thanks for your message. This definitely seems like something that should be in here. 

I'll try to get this added very soon.

**Comment by florianm** [comment]
@sckott would you mind if we submitted a pull request?

**Comment by sckott** [comment]
@florianm oh yes, of course

**Comment by sckott** [comment]
@florianm hi, it's been a while, should I do this?

**Comment by florianm** [comment]
@sckott sorry we got delayed with sick leave, should know by today, stay tuned!

**Comment by sckott** [comment]
@florianm okay, thx

**Comment by florianm** [comment]
Implemented by #31 (function) and #32 (tests) - thanks for the merge @sckott !

**Comment by sckott** [comment]
thank you
